tracing-appender/src/rolling.rs

```diff
@@ -1539,49 +1228,109 @@ mod test {
     }
 
     #[test]
-    fn size_based_rotation() {
+    fn test_max_log_size() {
+        use std::sync::{Arc, Mutex};
+        use tracing_subscriber::prelude::*;
+
+        let format = format_description::parse(
+            "[year]-[month]-[day] [hour]:[minute]:[second] [offset_hour \
+         sign:mandatory]:[offset_minute]:[offset_second]",
+        )
+            .unwrap();
+
+        let now = OffsetDateTime::parse("2020-02-01 10:01:00 +00:00:00", &format).unwrap();
         let directory = tempfile::tempdir().expect("failed to create tempdir");
+        let (state, writer) = Inner::new(
+            now,
+            Rotation::HOURLY.with_max_bytes(10),
+            directory.path(),
+            None,
+            None,
+            None,
+        )
+            .unwrap();
 
-        let mut appender = RollingFileAppender::builder()
-            .rotation(Rotation::max_bytes(8))
-            .filename_prefix("size_based_rotation")
-            .filename_suffix("log")
-            .build_with_now(&directory, Box::new(|| OffsetDateTime::UNIX_EPOCH))
-            .expect("initializing rolling file appender failed");
+        let clock = Arc::new(Mutex::new(now));
+        let now = {
+            let clock = clock.clone();
+            Box::new(move || *clock.lock().unwrap())
+        };
+        let appender = RollingFileAppender { state, writer, now };
+        let default = tracing_subscriber::fmt()
+            .without_time()
+            .with_level(false)
+            .with_target(false)
+            .with_max_level(tracing_subscriber::filter::LevelFilter::TRACE)
+            .with_writer(appender)
+            .finish()
+            .set_default();
 
-        writeln!(appender, "(file1) more than 8 bytes").unwrap();
-        writeln!(appender, "(file2) more than 8 bytes again").unwrap();
-        writeln!(appender, "(file3) and here is a third file").unwrap();
+        // Log two lines that will exceed the max file size
+        tracing::info!("1.log");
+        tracing::info!("2.log-with-contents");
 
-        let dir_contents = fs::read_dir(&directory).expect("read directory");
-        println!("dir={:?}", dir_contents);
+        // Log another two lines exceeding the limit
+        tracing::info!("3.log");
+        tracing::info!("4.log-with-contents");
 
-        let expected_files = [
-            "size_based_rotation.1970-01-01.0.log",
-            "size_based_rotation.1970-01-01.1.log",
-            "size_based_rotation.1970-01-01.2.log",
-        ];
-        let mut expected_files: std::collections::HashSet<&str> =
-            expected_files.iter().copied().collect();
```

Comment:
I recommend tracking the list of expected files like before. The current test will not fail if one or even all files are missing.

Reply:
Good catch! I'll add that back.